addons/account/models/account_journal.py

```diff
@@ -548,7 +548,8 @@ def create_invoice_from_attachment(self, attachment_ids=[]):
         for attachment in attachments:
             invoice = self.env['account.move'].create({})
             attachment.write({'res_model': 'mail.compose.message'})
-            invoice.message_post(attachment_ids=[attachment.id])
+            invoice = self.env['account.edi'].search([])._create_invoice_from_attachment(attachment)
```

Comment:
Calling _create_invoice_from_attachment from here creates a circular dependency between account and account_edi